daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1795,15 +1838,28 @@ private[lf] object SBuiltin {
     }
   }
 
+  /** $crash :: Text -> Unit -> Nothing */
+  private[speedy] final case class SBCrash(reason: String) extends SBuiltin(1) {
+
+    override private[speedy] def execute(
+        args: util.ArrayList[SValue],
+        machine: Machine,
+    ): Control = {
+      getSUnit(args, 0)
+
+      crash(reason)
+    }
+  }
+
   /** $try-handler :: Optional (Token -> a) -> AnyException -> Token -> a (or re-throw) */
   final case object SBTryHandler extends SBuiltin(3) {
     override private[speedy] def execute(
         args: util.ArrayList[SValue],
         machine: Machine,
     ): Control = {
+      checkToken(args, 2)
       val opt = getSOptional(args, 0)
       val excep = getSAny(args, 1)
-      checkToken(args, 2)
```

Comment:
why changing the order of evaluation ?

Reply:
Left overs from a previous refactor attempt - I'll revert this change.